src/cmd/services/m3coordinator/downsample/samples_appender.go

```diff
@@ -150,7 +151,20 @@ func (a *samplesAppender) appendTimedSample(sample aggregated.Metric) error {
 }
 
 func (a *samplesAppender) emitMetrics() {
-	a.processedCounter.Inc(int64(len(a.stagedMetadatas)))
+	a.processedCountNonRollup.Inc(int64(len(a.stagedMetadatas)))
+
+	for _, metadata := range a.stagedMetadatas {
+		// For each metadata check the number of pipeline operations and add
+		// them as processed by rollup. If there are more than a single
+		// operation then consider them.
+		numOperations := 0
+		for _, pipeline := range metadata.Pipelines {
+			numOperations += len(pipeline.Pipeline.Operations)
+		}
+		if numOperations > 1 {
+			a.processedCountRollup.Inc(int64(numOperations - 1))
```

Comment:
Can we emit two metrics here?

One with `Inc(numOperations - 1)` but also one that is just `Inc(1)`?

Since I think in general whether a rollup has 2 or 3 transforms and then a rollup, or just straight to a rollup, we may want to treat the same. Emitting both versions would help us be able to have visibility on both use cases.